### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by serhii56465**

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -1 +1,56 @@
-# write your code here
+from unittest import mock
+from app.main import cryptocurrency_action
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_equal_current(prediction):
+    prediction.return_value = 10
+    assert cryptocurrency_action(10) == "Do nothing"
```
Comment (liuda-chuzdiuk) [inline tests/test_main.py:8]: well, this test failed because in this case cryptocurrency_action returns "Sell all your cryptocurrency"

**Review by liuda-chuzdiuk — CHANGES_REQUESTED** [review]

File: tests/test_main.py [hunk 2]
```diff
@@ -1 +1,56 @@
...
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_equal_zero(prediction):
+    prediction.return_value = 0
+    assert cryptocurrency_action(10) == "Sell all your cryptocurrency"
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_equal_1_0_6(prediction):
+    prediction.return_value = 1.06
+    assert cryptocurrency_action(10) == "Sell all your cryptocurrency"
```
Comment (MasterpieceElbow) [inline tests/test_main.py:56]: You don't have to create so many tests, group your tests by return value of `cryptocurrency_action`. Write few assertions in one tests

**Review by MasterpieceElbow — CHANGES_REQUESTED** [review]

File: tests/test_main.py [hunk 3]
```diff
@@ -1 +1,32 @@
...
+    prediction.return_value = 10
+    assert cryptocurrency_action(10) == "Do nothing"
+    prediction.return_value = 9.6
+    assert cryptocurrency_action(10) == "Do nothing"
+    prediction.return_value = 9.5
+    assert cryptocurrency_action(10) == "Do nothing"
+    prediction.return_value = 10.4
+    assert cryptocurrency_action(10) == "Do nothing"
+    prediction.return_value = 10.5
+    assert cryptocurrency_action(10) == "Do nothing"
```
Comment (MasterpieceElbow) [inline tests/test_main.py:16]: Add boundary conditions, you don't check when `get_exchange_rate_prediction` return value equals exactly 1.05 or 0.95 * value passed in `cryptocurrency_action`.

**Review by MasterpieceElbow — CHANGES_REQUESTED** [review]

**Review by MasterpieceElbow — CHANGES_REQUESTED** [review]
